### PR TITLE
RE2022-62 Add Simple Collection Matching UI

### PR DESCRIPTION
**Pull request by dauglyon**
- Adds matching UI pane to Collection Detail Page
- Adds matched views for the two current data products (genome attrbs, taxa count)
- **There may still be some bugs but hopefully I have squashed them all**
- along with matching v0, logged-in visibility issue fix, better (real) async tables, an (exponential) backoff polling util, better async select, and an updated workspace API client definition

Screen Shot 2023-03-15 at 4 56 46 PM
Screen Shot 2023-03-15 at 4 57 33 PM
Screen Shot 2023-03-15 at 4 57 40 PM
Screen Shot 2023-03-15 at 4 59 28 PM
Screen Shot 2023-03-15 at 4 59 40 PM

### CODE REVIEW

File: src/features/collections/MatchPane.tsx [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  const matchersQuery = getCollectionMatchers.useQuery(collectionId);
+  const matchers = matchersQuery.data?.data;
+  const [matcherSel, setMatcherSel] = useState<SelectOption | undefined>();
+  const matcherSelected = matchers?.find((d) => d.id === matcherSel?.value);
+  const matcherOptions: SelectOption[] = useMemo(() => {
+    setMatcherSel(undefined);
+    return (
+      matchers?.map((matcher) => ({
+        value: matcher.id,
+        label: `${snakeCaseToHumanReadable(matcher.id)}: ${
```
Comment (dauglyon) [inline src/features/collections/MatchPane.tsx:79]: re: @MrCreosote 

Don't use the label from the server, or add user-oriented labels server-side and use those here

Comment (dauglyon) [reply src/features/collections/MatchPane.tsx:79]: As label name/desc aren't user-facing atm

File: src/common/components/Select.test.tsx [hunk 2]
```diff
@@ -108,35 +109,33 @@ test('Multi select onchange and remove', async () => {
...
-    return [
-      { value: 'chocolate', label: 'Chocolate' },
-      { value: 'strawberry', label: 'Strawberry' },
-      { value: 'vanilla', label: 'Vanilla' },
-      { value: 'test-value', label: inputValue },
-    ];
+  const Parent = () => {
+    const [options, setOptions] = useState<SelectOption[]>([]);
+    const handleSuggest = (inputValue: string) => {
+      new Promise<void>((r) => {
```
Comment (dakotablair) [inline src/common/components/Select.test.tsx:115]: Try wrapping this in `act` to fix `console.error`s in the tests.

File: src/features/collections/data_products/TaxaCount.tsx [hunk 3]
```diff
@@ -25,15 +26,28 @@ export const TaxaCount: FC<{
...
+    [collection_id, matchId, rank?.value]
   );
-  const countsQuery = getTaxaCountRank.useQuery(countsParams, { skip: !rank });
-  const counts = countsQuery.data?.data || [];
-  const max = counts.reduce((max, { count }) => (max > count ? max : count), 0);
+  const [polling, setPolling] = useState(false);
+  useEffect(() => setPolling(!!countsParams.match_id), [countsParams]); // Poll for new queries with a matchId
+  const countsQuery = getTaxaCountRank.useQuery(countsParams, {
+    skip: !rank,
+    pollingInterval: polling ? 1000 : 0,
```
Comment (dakotablair) [inline src/features/collections/data_products/TaxaCount.tsx:38]: Consider using exponential backoff here.

File: src/features/collections/data_products/TaxaCount.tsx [hunk 4]
```diff
@@ -25,15 +26,28 @@ export const TaxaCount: FC<{
...
-    () => ({ collection_id, rank: String(rank?.value) }),
-    [collection_id, rank?.value]
+    () => ({ collection_id, rank: String(rank?.value), match_id: matchId }),
+    [collection_id, matchId, rank?.value]
   );
-  const countsQuery = getTaxaCountRank.useQuery(countsParams, { skip: !rank });
-  const counts = countsQuery.data?.data || [];
-  const max = counts.reduce((max, { count }) => (max > count ? max : count), 0);
+  const [polling, setPolling] = useState(false);
+  useEffect(() => setPolling(!!countsParams.match_id), [countsParams]); // Poll for new queries with a matchId
```
Comment (dakotablair) [inline src/features/collections/data_products/TaxaCount.tsx:35]: We should probably decide on a preferred style (operators versus explicit type casts) for things like
- `Boolean(val)`
- `!!val`
- `Number(val)`
- `+val`

File: src/features/params/hooks.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      });
+      navigate(path);
+    },
+    [loc.pathname, navigate, params]
+  );
+};
+
+export const useAppParam = <Key extends keyof ParamsState>(key: Key) => {
+  const val = useAppSelector((state) => state.params[key]);
+  return val as NonNullable<ParamsState[Key]> | undefined;
```
Comment (dakotablair) [inline src/features/params/hooks.ts:31]: This may be the source of the taxa count `match_id` error.

**Comment by dakotablair** [comment]
Please also merge the code from PR #61 into this PR.

**Review by dakotablair — APPROVED** [review]

**Pull request merged**